freesitemgr

```diff
@@ -427,7 +447,26 @@ def main():
     elif cmd == 'add':
         if not sitemgr.node:
             noNodeError(sitemgr, "Cannot add site")
-        addSite(sitemgr)
+        if args:
+            sitepath = args[0]
+            sitename = addSite(sitemgr, sitepath)
+        else:
+            sitename = addSite(sitemgr)
+    
+        if opts["update-after-add"]: 
+            # renew sitemgr, then update 
+            # FIXME: Move the whole if-elif-else into a function to
+            # make it clear that nothing spills out.
+            try:
+                sitemgr.node.shutdown()
+            except:
+                try:
+                    sitemgr.node.socket.close()
```

Comment:
That the node doesn't do this is concerning to me. The socket should be internal to it.